armi/reactor/parameters/parameterDefinitions.py

```diff
@@ -440,6 +440,8 @@ def extend(self, other):
             self._representedTypes
         )
         assert self is not other
+        if other is None:
+            raise ValueError(f"Cannot extend {self} with `None`.")
```

Comment:
don't try to extend self with none!

This was added specifically because the previous error message was indecipherable. Given this error message, it was very clear what was wrong and how to fix.